components/selectors/tree.rs

```diff
@@ -58,10 +58,21 @@ pub trait Element: Sized + Clone + Debug {
 
     fn is_html_element_in_html_document(&self) -> bool;
 
-    fn local_name(&self) -> &<Self::Impl as SelectorImpl>::BorrowedLocalName;
+    fn local_name_hash(&self) -> u32;
```

Comment:
This is ugly. I know. We need three methods instead of one now...

Reply:
I think we don't need `get_local_name` or `get_namespace`, instead `fn local_name` and `fn namespace` should move to `TElement` (in `components/style/dom.rs`).